### ISSUE TITLE
Can't authenticate to admin console

### ISSUE THREAD

**Issue by dguertin** [cazzerson/Suma#63]
I've installed Suma on a Red Hat Linux 7 server running Apache 2.4.6. I've followed the steps in the installation page, and browsing to http://<sumaserver>/sumaserver shows success.

http://<sumaserver>/sumaserver/admin brings me to the Suma Administrative Tools login page, but when I log in, I receive "500 Application error". The credentials I'm using are set in \<suma_install_dir\>/service/config/config.yaml:

>     production:
>     sumaserver:
>         db:
>             host: localhost
>             platform: Pdo_Mysql
>             dbname: production
>             user: <suma user>
>             pword: <suma user password>
>             port: 3306
>         log:
>             path:
>             name: sumaserver.log
>         admin:
>             user: <admin user>
>             pass: <admin user password>
>     queryserver:
>         db:
>             limit: 10000
> 
>     development:
>     _extends: production
>     sumaserver:
>         db:
>             dbname: testing
>         log:
>             path:
> 

Both sets of credentials work in MySQL, i.e. I can access both databases outside of Suma.

What am I getting wrong?

**Comment by bretdavidson** [comment]
Hi,

Thanks for reaching out about Suma! I suspect this is an issue with the sumaserver.log. First, the config file requires a path for the log and it appears as if your config doesn't include a value for that configuration property. Second, the log needs to be writable by the web server and may require a permissions adjustment.

Hope that helps!

Best,

Bret

**Comment by dguertin** [comment]
Thanks for the suggestion. Unfortunately, it hasn't helped. I added the missing lines to config.yaml:

            log:
            path: /var/log/suma
            name: sumaserver.log

and restarted Apache, but I still just get "500 Application error", with no other info, and nothing in the syslog or Apache logs. Just to be sure, I temporarily set permissions on /var/log/suma to 777, but the results were the same.

**Comment by bretdavidson** [comment]
Two things:

1) Try turning on debugging in the service/web/config configuration file.

2) Could you send me a copy of your config files with the sensitive values removed (like database, etc.) to bddavids@ncsu.edu?

Thanks,

Bret

**Comment by dguertin** [comment]
Figured it out after turning on debugging. I saw this message:

"Configuration file (service/config/config.yaml) does not exist or is not readable."

The file was definitely there, but I had set the permissions to 640 root:root because it contains the database passwords in clear text. That meant that Apache couldn't read the file either, and I didn't realize at first that Apache needs to be able to read this file. I changed it to 640 root:apache, and now it's working.

Thanks,
Dave
